### PR TITLE
Define code conventions for our project 

### LINKED ISSUE #30: (SP): Define code conventions for our project

### PR DESCRIPTION
**Pull request by mhevyk**

### CODE REVIEW

**Comment by OliaHrytsak** [comment]
Maybe add aditional paragpapf like this?

image

**Comment by mhevyk** [comment]
> Maybe add aditional paragpapf like this?
> 
> image

I didn't add nothing related to detailed structure because I'm not sure about it. Maybe it should be discussed with our expert, what do you think?

File: README.md [hunk 1]
```diff
@@ -1,16 +1,225 @@
...
+| `npm run test:mutation`    | Runs mutation tests                      |
+| `npm run typescript:check` | Checks the validity of types             |
+| `npm run lint`             | Lints the project files                  |
+| `npm run lint:fix`         | Fixes lint errors in the project         |
+
+## Git flow
+
+### Pull requests
+
+- Use `git rebase` to merge brances locally.
```
Comment (OliaHrytsak) [inline README.md]: branches

File: README.md [hunk 2]
```diff
@@ -1,16 +1,225 @@
...
+
+Use pattern: `<prefix>/<issue_id>/<short_description>`
+
+For example: `feature/41/navbar`
+
+Allowed prefixes are:
+
+- `feature` - for new feature
+- `hotfix` - for urgent fixes
+- `bugfix` - for fixing existing functionality
```
Comment (OliaHrytsak) [inline README.md:69]: 'tests' also can be I think

File: README.md [hunk 3]
```diff
@@ -1,16 +1,225 @@
...
+### Steps to create wrapper around mui component
+
+For example we want to create `Button` component. The following steps should be followed:
+
+1. Create folder, prefixed with `app`: `app-button`
+2. Create file prefixed with `App`: `AppButton.tsx`
+3. Create scss file for component: `AppButton.scss`
+4. Import scss file into `AppButton.tsx` using import alias: `import "@/components/app-button/AppButton.scss";`
+5. Import all scss variables from design system: `@import "@design-system";`
+6. Create base css class, prefixed with `spa`: `.spa-button`:
```
Comment (OliaHrytsak) [inline README.md:174]: Do we need this 'spa'?

File: README.md (hunk 1 above)
Comment (mhevyk) [reply README.md]: Thanks a lot, fixed

File: README.md (hunk 2 above)
Comment (mhevyk) [reply README.md:69]: Added test branch prefix. We can use it for unit, mutation and e2e tests

File: README.md (hunk 3 above)
Comment (mhevyk) [reply README.md:174]: `spa` prefix is needed to create our custom design system with prefix of our company. Also it allows avoid collisions with other third-party css packages or another classes in general

**Comment by D1nviz** [comment]
I think it will be nice to have info about organizing imports. 
For example:
1. Imports from packages
2. Imports like components, containers
3. types, constants, hooks etc.
4. styles

**Review by OliaHrytsak — APPROVED** [review]

**Review by heshvab — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

**Review by D1nviz — APPROVED** [review]

**Pull request merged**